open_spiel/games/gin_rummy_test.cc

```diff
@@ -532,6 +533,32 @@ void OklahomaTest() {
   SPIEL_CHECK_EQ(returns[0], -88);
 }
 
+// Basic Observer functionality test.
+void ObserverTest() {
+  GameParameters params;
+  std::shared_ptr<const open_spiel::Game> game =
+      open_spiel::LoadGame("gin_rummy", params);
+
+  std::shared_ptr<Observer> observer = game->MakeObserver(kDefaultObsType, params);
+  Observation observation = Observation(*game, observer);
+
+  std::unique_ptr<open_spiel::State> state = game->NewInitialState();
+  std::vector<Action> initial_actions;
+  initial_actions = {1,  4,  5,  6,  17, 18, 19, 30, 31, 32, 2,  3,
+                     16, 29, 43, 44, 45, 7,  20, 33, 0,  52, 55, 1};
+  for (auto action : initial_actions) state->ApplyAction(action);
+  std::cout << state->ToString() << std::endl;
+
+  std::cout << observation.StringFrom(*state, 0) << std::endl;
```

Comment:
Can be simplified to `for(int player = 0; player < 2; player++) {` ...

Reply:
Fixed.